openbas-model/src/main/java/io/openbas/database/repository/ExerciseRepository.java

```diff
@@ -20,170 +20,244 @@ public interface ExerciseRepository extends CrudRepository<Exercise, String>,
+          "LEFT JOIN exercises_teams ext ON ext.exercise_id = ex.exercise_id " +
+          "LEFT JOIN pauses ON pauses.pause_exercise = ex.exercise_id " +
+          "LEFT JOIN exercises_documents ed ON ed.exercise_id = ex.exercise_id " +
+          "LEFT JOIN articles art ON art.article_exercise = ex.exercise_id " +
+          "LEFT JOIN lessons_categories lc ON lc.lessons_category_exercise = ex.exercise_id " +
+          "LEFT JOIN lessons_questions lq ON lq.lessons_question_category = lc.lessons_category_id " +
+          "LEFT JOIN lessons_answers la ON la.lessons_answer_question = lq.lessons_question_id " +
+          "LEFT JOIN scenarios_exercises se ON se.exercise_id = ex.exercise_id " +
+          "LEFT JOIN users_teams ut ON ext.team_id = ut.team_id " +
+          "LEFT JOIN logs ON logs.log_exercise = ex.exercise_id " +
+          "WHERE ex.exercise_id = :exerciseId " +
+          "GROUP BY ex.exercise_id, inj.inject_scenario, se.scenario_id ;", nativeQuery = true)
+  RawExercise rawDetailsById(@Param("exerciseId") String exerciseId);
+
+  @Query(value =
+      " SELECT ex.exercise_id, "
+          + "ex.exercise_status, "
+          + "ex.exercise_start_date, "
+          + "ex.exercise_updated_at, "
+          + "ex.exercise_end_date, "
+          + "ex.exercise_name, "
+          + "ex.exercise_category, "
+          + "ex.exercise_subtitle, "
+          + " array_agg(distinct ie.inject_id) FILTER ( WHERE ie.inject_id IS NOT NULL ) as inject_ids, "
+          + " array_agg(distinct et.tag_id) FILTER ( WHERE et.tag_id IS NOT NULL ) as exercise_tags "
+          + "FROM exercises ex "
+          + "LEFT JOIN exercises_tags et ON et.exercise_id = ex.exercise_id "
+          + "LEFT JOIN injects_expectations ie ON ex.exercise_id = ie.exercise_id "
+          + "WHERE ex.exercise_id = :id "
+          + "GROUP BY ex.exercise_id ;", nativeQuery = true)
+  RawExercise rawById(@Param("id") String id);
+
+  @Query(value =
+      " SELECT ex.exercise_id, "
+          + "ex.exercise_status, "
+          + "ex.exercise_start_date, "
+          + "ex.exercise_updated_at, "
+          + "ex.exercise_end_date, "
+          + "ex.exercise_name, "
+          + "ex.exercise_category, "
+          + "ex.exercise_subtitle, "
+          + " array_agg(distinct ie.inject_id) FILTER ( WHERE ie.inject_id IS NOT NULL ) as inject_ids, "
+          + " array_agg(distinct et.tag_id) FILTER ( WHERE et.tag_id IS NOT NULL ) as exercise_tags "
+          + "FROM exercises ex "
+          + "LEFT JOIN scenarios_exercises s ON s.exercise_id = ex.exercise_id "
+          + "LEFT JOIN exercises_tags et ON et.exercise_id = ex.exercise_id "
+          + "LEFT JOIN injects_expectations ie ON ex.exercise_id = ie.exercise_id "
+          + "WHERE s.scenario_id IN (:scenarioIds) "
+          + "GROUP BY ex.exercise_id ;", nativeQuery = true)
+  List<RawExercise> rawAllByScenarioId(@Param("scenarioIds") List<String> scenarioIds);
```

Comment:
Same here, the query is not aligned with properties on RawExercise.
It's a big one